tests/foreman/api/test_contentviewversion.py

```diff
@@ -33,960 +33,1024 @@
+    # environments.
+    version = version.read()
+    assert len(version.environment) == 2
+
+
+@pytest.mark.tier2
+def test_positive_promote_out_of_sequence_environment(module_org, setUp):
+    """Promote a content view version to a lifecycle environment
+    that is 'out of sequence'.
+
+    :id: e88405de-843d-4279-9d81-cedaab7c23cf
+
+    :expectedresults: The promotion succeeds.
+
+    :CaseLevel: Integration
+    """
+    # Create a new content view...
+    _, lce2, _ = setUp
+    cv = entities.ContentView(organization=module_org).create()
+    # ... and publish it.
+    cv.publish()
+    # Refresh the entity
+    cv = cv.read()
+    # Check that we have a new version
+    assert len(cv.version) == 1
+    version = cv.version[0].read()
+    # The immediate lifecycle is lce1, not lce2
+    promote(version, lce2.id, force=True)
+    # Assert that content view version is found in 2 lifecycle
+    # environments.
+    version = version.read()
+    assert len(version.environment) == 2
+
+
+@pytest.mark.tier2
+def test_negative_promote_valid_environment(setUp):
+    """Promote the default content view version.
+
+    :id: cd4f3c3d-93c5-425f-bc3b-d1ac17696a4a
+
+    :expectedresults: The promotion fails.
+
+    :CaseLevel: Integration
+
+    :CaseImportance: Low
+    """
+    lce1, _, default_cv = setUp
+    with pytest.raises(HTTPError):
+        promote(default_cv, lce1.id)
+
+
+@pytest.mark.tier2
+def test_negative_promote_out_of_sequence_environment(setUp, module_org):
+    """Promote a content view version to a lifecycle environment
+    that is 'out of sequence'.
+
+    :id: 621d1bb6-92c6-4209-8369-6ea14a4c8a01
+
+    :expectedresults: The promotion fails.
+
+    :CaseLevel: Integration
+    """
+    # Create a new content view...
+    _, lce2, _ = setUp
+    cv = entities.ContentView(organization=module_org).create()
+    # ... and publish it.
+    cv.publish()
+    # Refresh the entity
+    cv = cv.read()
+    # Check that we have a new version
+    assert len(cv.version) == 1
+    version = cv.version[0].read()
+    # The immediate lifecycle is lce1, not lce2
+    with pytest.raises(HTTPError):
+        promote(version, lce2.id)
+
+
+# Tests for content view version promotion.
+
+
+@pytest.mark.tier2
+@pytest.mark.skipif((not settings.repos_hosting_url), reason='Missing repos_hosting_url')
+def test_positive_delete():
+    """Create content view and publish it. After that try to
+    disassociate content view from 'Library' environment through
+    'delete_from_environment' command and delete content view version from
+    that content view. Add repository and gpg key to initial content view
+    for better coverage
+
+    :id: 066dec47-c942-4c01-8956-359c8b23a6d4
+
+    :expectedresults: Content version deleted successfully
+
+    :CaseLevel: Integration
+
+    :CaseImportance: Critical
+    """
+    key_content = read_data_file(ZOO_CUSTOM_GPG_KEY)
+    org = entities.Organization().create()
```

Comment:
```suggestion
```
Try not to create a new org as it takes time and places another load on Satellite under test